lib/agent0/agent0/hyperdrive/policies/random_hold.py

```diff
@@ -0,0 +1,313 @@
+                elif position.action_type == HyperdriveActionType.OPEN_SHORT:
+                    short_ready_to_close = True
+                else:
+                    # Sanity check
+                    raise ValueError(f"Action type {position.action_type} not in allowable actions")
+
+        if wallet.balance.amount <= pool_state.pool_config.minimum_transaction_amount:
+            all_available_actions = []
+        else:
+            all_available_actions = [
+                HyperdriveActionType.ADD_LIQUIDITY,
+            ]
+            # We hard cap the number of open positions to keep track of
+            if len(self.open_positions) < self.max_open_positions:
+                all_available_actions.extend(
+                    [
+                        HyperdriveActionType.OPEN_LONG,
+                        HyperdriveActionType.OPEN_SHORT,
+                    ]
+                )
+        if long_ready_to_close:  # if the agent has longs ready to close
+            all_available_actions.append(HyperdriveActionType.CLOSE_LONG)
+        if short_ready_to_close:  # if the agent has shorts ready to close
+            all_available_actions.append(HyperdriveActionType.CLOSE_SHORT)
+        if wallet.lp_tokens:
+            all_available_actions.append(HyperdriveActionType.REMOVE_LIQUIDITY)
+        if wallet.withdraw_shares and pool_state.pool_info.withdrawal_shares_ready_to_withdraw > 0:
+            all_available_actions.append(HyperdriveActionType.REDEEM_WITHDRAW_SHARE)
+        # down select from all actions to only include allowed actions
+        return [action for action in all_available_actions if action in self.allowable_actions]
+
+    def close_random_long(
+        self, interface: HyperdriveReadInterface, wallet: HyperdriveWallet
+    ) -> list[Trade[HyperdriveMarketAction]]:
+        """Closes a random long that's ready to be closed.
+
+        Arguments
+        ---------
+        interface: HyperdriveReadInterface
+            Interface for the market on which this agent will be executing trades (MarketActions).
+        wallet: HyperdriveWallet
+            The agent's wallet.
+
+        Returns
+        -------
+        list[Trade[HyperdriveMarketAction]]
+            A list with a single Trade element for closing a Hyperdrive short.
+        """
+        # We scan open positions and select a long that's ready to be closed
+        longs_ready_to_close: list[RandomHold._Position] = [
+            position
+            for position in self.open_positions
+            if position.ready_to_close and position.action_type == HyperdriveActionType.OPEN_LONG
+        ]
+        # Sanity check, we should have at least one, otherwise close long wouldn't be an available action
+        assert len(longs_ready_to_close) > 0
+
+        # Select a random one
+        long_to_close = longs_ready_to_close[self.rng.integers(len(longs_ready_to_close))]
+        # Set flag that the transaction was sent
```

Comment:
why?  so we don't send it again?

Reply:
This flag is for bookkeeping, the end of `post_action` removes positions with transactions that were sent (after making sure they succeed)